kernel/fatfs.c

```diff
@@ -124,10 +124,41 @@ STATIC void init_direntry(struct dirent *dentry, unsigned attrib,
 int dos_open(char *path, unsigned flags, unsigned attrib, int fd)
 {
   REG f_node_ptr fnp = sft_to_fnode(fd);
-  int status = find_fname(path, D_ALL | attrib, fnp);
+  int status;
+
+  /* Special handling for volume name (they can coexist with files/directories) */
+  if (((attrib & (D_DIR | D_VOLID | D_SYSTEM | D_HIDDEN | D_RDONLY)) == D_VOLID) &&
+      (flags & O_TRUNC))
+  {
+    int numslashes;
+    char *p;
+    int ret;
+    char rootfiles[] = "?:\\????????.???";
+
+    /* Count the slashes to see if we are in the root directory */
+    for (p = path, numslashes = 0; *p; p++)
+      if (*p == '\\')
+        if (++numslashes > 1)
```

Comment:
Can't we combine these 2 checks here into one?

```
if (*p == '\\' && ++numslashes > 1)
    ...
```

Reply:
I tend to dislike pre/post operators with more than very simple logic, but if it's a deal breaker we could.

Reply:
removed loop altogether, thanks.